test/e2e/gameserver_test.go

```diff
@@ -205,7 +208,27 @@ func TestGameServerUnhealthyAfterDeletingPod(t *testing.T) {
 	err = podClient.Delete(pod.ObjectMeta.Name, nil)
 	assert.NoError(t, err)
 
-	_, err = framework.WaitForGameServerState(readyGs, v1alpha1.GameServerStateUnhealthy, 10*time.Second)
+	// TODO [markmandel@google.com]: Should GameServers that are Unhealthy and not in a fleet be deleted?
+	err = wait.PollImmediate(2*time.Second, time.Minute, func() (bool, error) {
+		gs, err := framework.AgonesClient.StableV1alpha1().GameServers(readyGs.Namespace).Get(readyGs.ObjectMeta.Name, metav1.GetOptions{})
+
+		// just in case
+		if k8serrors.IsNotFound(err) {
+			return true, nil
+		}
+
+		if err != nil {
+			logrus.WithError(err).Warn("error retrieving gameserver")
+			return false, nil
```

Comment:
Should this be `return false, err`? The func returns `(bool, error)` but none of the cases actually return a non-nil error.

Reply:
That's a good question - looking at the code, if any error is returned from the function, then `PollImmeadiate` will fail, and return an error - in this case, I don't think we actually want that to happen, we want it to retry and try again if something weird happens - because it may be a timing issue or self resolve?